src/server.cpp

```diff
@@ -1101,15 +1102,16 @@ void Server::Receive(float timeout)
 		pkt.clear();
 		peer_id = 0;
 		try {
+			// Round up since the target step length is the minimum step length,
+			// we only have millisecond precision and we don't want to busy-wait by calling
+			// ReceiveTimeoutMs(.., 0) repeatedly.
```

Comment:
since we have a float anyway this would be more straightforward:
`const u32 cur_timeout = std::ceil(remaining_time_us() / 1000.0f);`